tensorly/decomposition/cmtf_als.py

```diff
@@ -0,0 +1,157 @@
+import numpy as np
+
+import tensorly as tl
+from ..tenalg import khatri_rao
+from ..random import random_kruskal
+from ..tenalg import solve_least_squares
+
+
+# Authors: Isabell Lehmann <isabell.lehmann94@outlook.de>
+
+# License: BSD 3 clause
+
+
+def factor_match_score_3d(kruskal_tensor_3d_true, kruskal_tensor_2d_true, kruskal_tensor_3d_pred,
+                          kruskal_tensor_2d_pred):
+    """
+    Calculates factor match score (FMS) for comparing true and estimated matrices.
+
+    Assume a 3rd order tensor and a matrix are coupled in the 1th mode with
+    X = [[A, B, C]] and Y = [[A, V]]
+    The FMS shows how close the estimation of A, B, C, V is to the true matrices.
+    An FMS of 1 is a perfect match. The worst value is 0.
+
+    Formula for FMS according to Equation 4
+    @article{acar2011all,
+      title={All-at-once optimization for coupled matrix and tensor factorizations},
+      author={Acar, Evrim and Kolda, Tamara G and Dunlavy, Daniel M},
+      journal={arXiv preprint arXiv:1105.3422},
+      year={2011}
+    }
+
+    Parameters
+    ----------
+    kruskal_tensor_3d_true : Kruskal tensor
+        true tensor X = [[A, B, C]]
+    kruskal_tensor_3d_true : Kruskal tensor
+        true matrix Y = [[A, V]]
+    kruskal_tensor_3d_pred : Kruskal tensor
+        tensor consisting of estimated factor matrices A_pred, B_pred, C_pred: X_pred = [[A_pred, B_pred, C_pred]]
+    kruskal_tensor_3d_pred : Kruskal tensor
+        matrix consisting of estimated factor matrices A_pred, V_pred: Y_pred = [[A_pred, V_pred]]
+
+    Returns
+    -------
+    factor match score : float
+        value between 0 and 1 that shows how good the estimation is, 1 is perfect
+
+    """
+
+    lambda_, [a, B, C] = tl.kruskal_tensor.kruskal_normalise(kruskal_tensor_3d_true)
+    alpha_, [A, V] = tl.kruskal_tensor.kruskal_normalise(kruskal_tensor_2d_true)
+    xi = lambda_ + alpha_
+
+    lambda_pred, [a_pred, B_pred, C_pred] = tl.kruskal_tensor.kruskal_normalise(kruskal_tensor_3d_pred)
+    alpha_pred, [A_pred, V_pred] = tl.kruskal_tensor.kruskal_normalise(kruskal_tensor_2d_pred)
+    xi_pred = lambda_pred + alpha_pred
+
+    # take care of permutation of normalized matrices (not mentioned in paper, but necessary)
+    perm = (A.T @ A_pred) * (B.T @ B_pred) * (C.T @ C_pred)
+    order = tl.argmax(perm, axis=1)
+    perm = perm[:, order]  # perm[r,r] now corresponds to \prod_i (a^i_r.T @ ahat^i_r) / (||(a^i_r|| ||ahat^i_r||)
+    V_pred = V_pred[:, order]
+    xi_hat = xi_pred[order]
+
+    # calculate factor match score
+    rank = A.shape[1]
+    FMS = tl.zeros(rank)
+    for r in range(rank):
+        FMS[r] = (1 - tl.abs(xi[r] - xi_hat[r]) / tl.max([xi[r], xi_hat[r]])) * tl.abs(
+            perm[r, r] * tl.dot(V[:, r], V_pred[:, r]))
+
+    return tl.min(FMS)
+
+
+def cmtf_als_for_third_order_tensor(tensor_3d, matrix, rank):
```

Comment:
Naming is tough.. Assuming people will be familiar with that, we could simply use `cmtf_als_3d` ?
Though I think it might clearest to spell out `coupled_matrix_tensor_decomposition` `coupled_matrix_tensor_3d_decomposition` ?